crates/core/tedge_mapper/src/c8y/alarm_converter.rs

```diff
@@ -47,10 +47,16 @@ impl AlarmConverter {
             }
             Self::Synced => {
                 //Regular conversion phase
-                let tedge_alarm = ThinEdgeAlarm::try_from(
-                    input_message.topic.name.as_str(),
-                    input_message.payload_str()?,
-                )?;
+                let mqtt_payload = input_message.payload_str()?;
+                let mqtt_topic = input_message.topic.name.clone();
```

Comment:
Minor. I would avoid to clone the topic and to extract the payload twice even when there is no error.